need better reporting for files mentioned in Dockerfile but missing on disk

maybe I'm creating a duplicate, because we already had one instance

when sandbox.py or generator for tekton pipelines find that there is missing file, they crash with exception

turns out developers then have trouble figuring out it's their mistake and not the tools

Here's an example of the hard to understand error

https://github.com/opendatahub-io/notebooks/actions/runs/13951200220/job/39051215307?pr=943#step:19:61

```
+ /home/runner/work/notebooks/notebooks/scripts/sandbox.py --dockerfile jupyter/datascience/ubi9-python-3.11/Dockerfile.cpu -- podman build --cache-from ghcr.io/opendatahub-io/notebooks/workbench-images/build-cache --tag ghcr.io/opendatahub-io/notebooks/workbench-images:jupyter-datascience-ubi9-python-3.11-943_merge_78943fcdb427d052be1f1edd281db45e867cb4c9 --file jupyter/datascience/ubi9-python-3.11/Dockerfile.cpu '{};'
2025/03/19 16:18:50 found paths: ["jupyter/datascience/ubi9-python-3.11/Pipfile.lock","jupyter/datascience/ubi9-python-3.11/mongodb-org-6.0.repo-x86_64","jupyter/datascience/ubi9-python-3.11/mssql-2022.repo-x86_64","jupyter/datascience/ubi9-python-3.11/runtime-images","jupyter/datascience/ubi9-python-3.11/setup-elyra.sh","jupyter/datascience/ubi9-python-3.11/utils","jupyter/minimal/ubi9-python-3.11/start-notebook.sh","jupyter/utils"]
__file__='/home/runner/work/notebooks/notebooks/scripts/sandbox.py' started with args=Namespace(dockerfile=PosixPath('jupyter/datascience/ubi9-python-3.11/Dockerfile.cpu'), remaining=['--', 'podman', 'build', '--cache-from', 'ghcr.io/opendatahub-io/notebooks/workbench-images/build-cache', '--tag', 'ghcr.io/opendatahub-io/notebooks/workbench-images:jupyter-datascience-ubi9-python-3.11-943_merge_78943fcdb[42](https://github.com/opendatahub-io/notebooks/actions/runs/13951200220/job/39051215307?pr=943#step:19:43)7d052be1f1edd281db45e867cb4c9', '--file', 'jupyter/datascience/ubi9-python-3.11/Dockerfile.cpu', '{};'])
prereqs=[PosixPath('jupyter/datascience/ubi9-python-3.11/Pipfile.lock'), PosixPath('jupyter/datascience/ubi9-python-3.11/mongodb-org-6.0.repo-x86_64'), PosixPath('jupyter/datascience/ubi9-python-3.11/mssql-2022.repo-x86_64'), PosixPath('jupyter/datascience/ubi9-python-3.11/runtime-images'), PosixPath('jupyter/datascience/ubi9-python-3.11/setup-elyra.sh'), PosixPath('jupyter/datascience/ubi9-python-3.11/utils'), PosixPath('jupyter/minimal/ubi9-python-3.11/start-notebook.sh'), PosixPath('jupyter/utils')]
Traceback (most recent call last):
  File "/home/runner/work/notebooks/notebooks/scripts/sandbox.py", line 82, in <module>
    sys.exit(main())
             ~~~~^^
  File "/home/runner/work/notebooks/notebooks/scripts/sandbox.py", line 44, in main
    setup_sandbox(prereqs, pathlib.Path(tmpdir))
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/notebooks/notebooks/scripts/sandbox.py", line 78, in setup_sandbox
    shutil.copy(dep, tmpdir / dep.parent)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/linuxbrew/.linuxbrew/Cellar/python@3.13/3.13.2/lib/python3.13/shutil.py", line 428, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
    ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/linuxbrew/.linuxbrew/Cellar/python@3.13/3.13.2/lib/python3.13/shutil.py", line 260, in copyfile
    with open(src, 'rb') as fsrc:
         ~~~~^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'jupyter/datascience/ubi9-python-3.11/runtime-images'
make: *** [Makefile:128: jupyter-datascience-ubi9-python-3.11] Error 1
Error: Process completed with exit code 2.
```

--- a/scripts/sandbox.py
+++ b/scripts/sandbox.py
@@ -100,6 +100,11 @@ def setup_sandbox(prereqs: list[pathlib.Path], tmpdir: pathlib.Path):
     for dep in prereqs:
         if dep.is_absolute():
             dep = dep.relative_to(ROOT_DIR)
+
+        if not dep.exists():
+            logging.error(f"File or directory '{dep}' referenced in the Dockerfile was not found on disk. Please ensure the file exists.")
+            sys.exit(1)
+
         if dep.is_dir():
             shutil.copytree(dep, tmpdir / dep, symlinks=False, dirs_exist_ok=True)
         else:

